src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -2596,9 +2596,10 @@ int W3DModelDraw::Get_Current_Bone_Positions(
 void W3DModelDraw::React_To_Transform_Change(const Matrix3D *matrix, const Coord3D *pos, float angle)
 {
     if (m_renderObject) {
-        Matrix3D m(Get_Drawable()->Get_Transform_Matrix());
-        Adjust_Transform_Mtx(m);
-        m_renderObject->Set_Transform(m);
+        const Matrix3D *dtm = Get_Drawable()->Get_Transform_Matrix();
+        Matrix3D rtm(*dtm);

```

Comment:
how did this fix it?

Reply:
I take it there was a different constructor being called when passed a pointer instead of a reference? That suggests there could easily be other issues elsewhere in the codebase and a massive footgun

Reply:
We already checked for other possible similar issues and found none. And we are looking at warnings that will prevent such an issue in the future.